beta-src/src/state/game/game-api-slice.ts

```diff
@@ -293,13 +305,116 @@ const drawBuilds = (state) => {
+          val.countryID === context.countryID &&
+          val.countryID !== val.ownerCountryID
+        );
+      });
+
+      let userDisbandingUnits;
+      let territoryKeys;
+      console.log("ot", overtakenTerritories);
+
+      if (overtakenTerritories && overtakenTerritories.length > 0) {
+        const terrKeyTempArray: string[] = [];
+        const disbandingUnitsTempArray: CurrentOrder[] = [];
+        overtakenTerritories.forEach(([terrKey, val]) => {
+          const { unitID } = val;
+          console.log("OM", ordersMeta);
+          const overtakenUnitsInOrder = currentOrders.find(
+            (o) =>
+              (val.id === units[o.unitID].terrID &&
+                o.unitID !== unitID &&
+                o.type === "Disband") ||
+              (val.id === units[o.unitID].terrID &&
+                o.unitID !== unitID &&
+                ordersMeta[o.id] &&
+                ordersMeta[o.id].update.type === "Disband"),
+          );
+
+          console.log("otuio", overtakenUnitsInOrder);
+
+          if (overtakenUnitsInOrder) {
+            disbandingUnitsTempArray.push(overtakenUnitsInOrder);
+            terrKeyTempArray.push(terrKey);
+          }
+        });
+        territoryKeys = terrKeyTempArray;
+        userDisbandingUnits = disbandingUnitsTempArray;
+      }
+      if (!userDisbandingUnits) {
+        userDisbandingUnits = currentOrders;
+      }
+      console.log("userDisbandingUnits", userDisbandingUnits);
+      if (userDisbandingUnits) {
+        userDisbandingUnits.forEach((order, index) => {
+          console.log(territoryKeys);
+          if (
+            context.orderStatus === "Saved,Completed,Ready" &&
+            order.type === "Disband" &&
+            territoryKeys
+          ) {
+            const command: GameCommand = {
+              command: "SET_UNIT",
+              data: {
+                setUnit: {
+                  componentType: "Icon",
+                  country: undefined,
+                  iconState: UIState.DISBANDED,
+                  unitSlotName: "main",
+                  unitType: undefined,
+                },
+              },
+            };
+
+            setCommand(
+              state,
+              command,
+              "territoryCommands",
+              Territory[territoryKeys[index]],
+            );
+          } else if (
+            context.orderStatus === "Saved,Completed" &&
+            order.type === "Disband"
+          ) {
+            const command: GameCommand = {
+              command: "DISBAND",
+            };
+            console.log("command", command);
+            setCommand(state, command, "unitCommands", order.unitID);
+          }
+        });
+      }
+    }
+  }
+};
+
 const gameApiSlice = createSlice({
   name: "game",
   initialState,
   reducers: {
     updateOrdersMeta(state, action: UpdateOrdersMetaAction) {
       updateOrdersMeta(state, action.payload);
       drawBuilds(state);
+      updateUnitsDisbanding(state);
```

Comment:
you should add this to the updateOrdersMeta function, around where the other function is that draws arrows.